libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -826,20 +826,25 @@ export class PackageExtractor {
         queue,
         async ([sourcePath, callback]: [string, () => void]) => {
           const relativeSourcePath: string = path.relative(sourceFolderPath, sourcePath);
-          if (
-            relativeSourcePath !== '' &&
-            (ignoreFilter.ignores(relativeSourcePath) || isFileExcluded(relativeSourcePath))
-          ) {
+          if (relativeSourcePath !== '' && ignoreFilter.ignores(relativeSourcePath)) {
             callback();
             return;
           }
 
           const sourcePathNode: PathNode = await state.symlinkAnalyzer.analyzePathAsync(sourcePath);
-
           if (sourcePathNode.kind === 'file') {
+            // Only ignore files and not folders to ensure that we traverse the contents of all folders. This is
+            // done so that we can match against subfolder patterns, ex. "src/subfolder/**/*"
+            if (relativeSourcePath !== '' && isFileExcluded(relativeSourcePath)) {
+              callback();
```

Comment:
Can this callback be converted to a promise?

Reply:
The callback is built into the AsyncQueue and is required to notify that the queue can be iterated. I'm open to suggestions on how the AsyncQueue class works for sure, but I'd call it out of scope for this PR.